src/esp/gfx/PBRDrawable.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "PBRDrawable.h"
+
+#include <Corrade/Containers/ArrayViewStl.h>
+#include <Corrade/Utility/FormatStl.h>
+
+namespace Mn = Magnum;
+
+namespace esp {
+namespace gfx {
+
+PBRDrawable::PBRDrawable(scene::SceneNode& node,
+                         Mn::GL::Mesh& mesh,
+                         ShaderManager& shaderManager,
+                         const Mn::ResourceKey& lightSetupKey,
+                         const Mn::ResourceKey& materialDataKey,
+                         DrawableGroup* group)
+    : Drawable{node, mesh, group},
+      shaderManager_{shaderManager},
+      lightSetup_{shaderManager.get<LightSetup>(lightSetupKey)},
+      materialData_{
+          shaderManager.get<MaterialData, PBRMaterialData>(materialDataKey)} {
+  updateShader().updateShaderLightParameters();
+}
+
+void PBRDrawable::setLightSetup(const Mn::ResourceKey& lightSetup) {
+  lightSetup_ = shaderManager_.get<LightSetup>(lightSetup);
+  updateShader().updateShaderLightParameters();
+}
+
+void PBRDrawable::draw(const Mn::Matrix4& transformationMatrix,
+                       Mn::SceneGraph::Camera3D& camera) {
+  // no need to call updateShaderLightParameters() in the draw loop,
+  // only need to update the position or direction of the lights
+  updateShader().updateShaderLightDirectionParameters(transformationMatrix,
+                                                      camera);
+
+  (*shader_)
+      // e.g., semantic mesh has its own per vertex annotation, which has been
+      // uploaded to GPU so simply pass 0 to the uniform "objectId" in the
+      // fragment shader
+      .setObjectId(
+          static_cast<RenderCamera&>(camera).useDrawableIds()
+              ? drawableId_
+              : (materialData_->perVertexObjectId ? 0 : node_.getSemanticId()))
+      .setTransformationMatrix(transformationMatrix)  // modelview matrix
+      .setMVPMatrix(camera.projectionMatrix() * transformationMatrix)
+      // because we actually do not have scaling, only rotation, so that the
+      // modelview matrix mv = inv(mv^T) where mv stands for the up-left 3x3
+      // part of the modelview matrix
+      .setNormalMatrix(transformationMatrix.rotationScaling())
```

Comment:
Why not use [normalMatrix()](https://doc.magnum.graphics/magnum/classMagnum_1_1Math_1_1Matrix4.html#aae3f4e7f98d6838ef7be4f70531a7583)? It's a *relatively* new thing, so you might not be aware of it yet. 

And it looks like `GenericDrawable` should receive [the same treatment](https://github.com/facebookresearch/habitat-sim/blob/6774fc4b12a06bf79eab6585606936004a643ca0/src/esp/gfx/GenericDrawable.cpp#L98).